packages/pwa-kit-react-sdk/src/ssr/universal/withLegacyGetProps.js

```diff
@@ -0,0 +1,49 @@
+import hoistNonReactStatic from 'hoist-non-react-statics'
+import {getAppConfig} from './compatibility'
+import {FetchStrategy} from './fetchStrategy'
+import React from 'react'
+
+export const withLegacyGetProps = (Wrapped) => {
+    const wrappedComponentName = Wrapped.displayName || Wrapped.name
+
+    class WithLegacyGetProps extends FetchStrategy {
+        render() {
+            return <Wrapped {...this.props} />
+        }
+
+        static async doInitAppState({App, match, route, req, res, location}) {
+            const AppConfig = getAppConfig()
+            const {params} = match
+
+            const components = [App, route.component]
+            const promises = components.map((c) =>
+                c.getProps
+                    ? c.getProps({
+                          req,
+                          res,
+                          params,
+                          location
+                      })
+                    : Promise.resolve({})
+            )
+
+            const [appProps, pageProps] = await Promise.all(promises)
+            return {
+                appProps,
+                pageProps,
+                __STATE_MANAGEMENT_LIBRARY: AppConfig.freeze(res.locals)
```

Comment:
In my mind the AppConfig's freeze/restore API's weren't part of the _getProps_ API. Should we stick this back in the `react-rendering` file. I'm guessing you pulled the __STATE_MANAGEMENT_LIBRARY freezing into this file because it looked cleaner no to mention it inside the react-rendering. But it might be a little weird restoring in one file and freezing in another. On top of that, you could end up not freezing if you don't use this HOC, but you still try to restore in the main and rendering files.

Reply:
Correction: Looks like your getAppConfig handles alway defining the freeze method. I still feel its better off back in the react-rendering file.